src/audio/copier/copier_host.c

```diff
@@ -6,12 +6,86 @@
 
 #include <sof/audio/component_ext.h>
 #include <sof/audio/host_copier.h>
+#include <sof/tlv.h>
 #include <sof/trace/trace.h>
 #include <ipc4/copier.h>
 #include <sof/audio/module_adapter/module/generic.h>
 
 LOG_MODULE_DECLARE(copier, CONFIG_SOF_LOG_LEVEL);
 
+#if CONFIG_HOST_DMA_STREAM_SYNCHRONIZATION
+struct fpi_sync_group {
+	uint32_t id;
+	uint32_t period;
+	uint32_t ref_count;
+	struct list_item item;
+};
+
+static struct list_item group_list_head = LIST_INIT(group_list_head);
+
+struct fpi_sync_group *find_group_by_id(uint32_t id)
```

Comment:
since it isn't in any header, please make `static`, this would also explain, that it's only used when adding and deleting groups, i.e. only in IPC thread context, so no locking is needed... supposedly

Reply:
should include all new added functions.